src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSHTMLFilePathToContentTypeProvider.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [FileExtension(RazorLSPConstants.CSHTMLFileExtension)]
```

Comment:
Purposefully not applying `[Shared]` because the infrastructure that spins up these `IFilePathToContentTypeProvider`s doesn't respect the `System.Composition` variant of MEF attributes (that's where that attribute is). I've found that it only gets constructed once anyhow.